packages/transaction-controller/src/TransactionController.ts

```diff
@@ -556,14 +621,117 @@ export class TransactionController extends BaseControllerV1<
 
     this.addPendingTransactionTrackerListeners();
 
+    // when transactionsController state changes
+    // check for pending transactions and start polling if there are any
+    this.subscribe(this.#checkForPendingTransactionAndStartPolling);
+
+    // TODO once v2 is merged make sure this only runs when
+    // selectedNetworkClientId changes
     onNetworkStateChange(() => {
       log('Detected network change', this.getChainId());
+      this.pendingTransactionTracker.onStateChange();
       this.onBootCleanup();
     });
 
     this.onBootCleanup();
+    this.messagingSystem.subscribe(
+      'NetworkController:stateChange',
+      (_, patches) => {
+        if (this.enableMultichain) {
+          const networkClients = this.getNetworkClientRegistry();
+          patches.forEach(({ op, path }) => {
+            if (op === 'remove' && path[0] === 'networkConfigurations') {
+              const networkClientId = path[1] as NetworkClientId;
+              delete networkClients[networkClientId];
+            }
+          });
+
+          this.#refreshTrackingMap(networkClients);
+          this.#refreshEtherscanRemoteTransactionSources(networkClients);
```

Comment:
Can this be invoked from within `#refreshTrackingMap` since this forms part of the tracking map and is only used here?

Plus since the remote transaction sources are created elsewhere, should this be called `#cleanEtherscanRemoteTransactionSources` or something similar?

Reply:
Thoughts on the rename of `#refreshEtherscanRemoteTransactionSources` to `#cleanEtherscanRemoteTransactionSources ?

Reply:
It doesn't just clean, it adds and removes based on the registry, same thing that refresh tracking map. I think we should stick with the `refresh` names since they have the same pattern.